parser/src/parser.cpp

```diff
@@ -225,24 +250,64 @@ int main(int argc, char* argv[])
   if (vm.count("force"))
     cc::util::removeTables(db, SQL_DIR);
 
-  cc::util::createTables(db, SQL_DIR);
+  if (vm.count("force") || isNewDb)
+    cc::util::createTables(db, SQL_DIR);
 
   //--- Start parsers ---//
 
   cc::parser::SourceManager srcMgr(db);
   cc::parser::ParserContext ctx(db, srcMgr, compassRoot, vm);
   pHandler.createPlugins(ctx);
 
+  // TODO: Handle errors returned by preparse().
+  std::vector<std::string> topologicalOrder = pHandler.getTopologicalOrder();
+  for (auto it = topologicalOrder.rbegin(); it != topologicalOrder.rend(); ++it)
+  {
+    LOG(info) << "[" << *it << "] preparse started!";
+    if(!pHandler.getParser(*it)->preparse())
+    {
+      LOG(error) << "[" << *it << "] preparse failed!";
+      return 2;
+    }
+  }
+
+  // TODO: Consider whether there is a better place for this code.
+  (cc::util::OdbTransaction(ctx.db))([&]
```

Comment:
Instead of trying to solve vexparse with Lisp, use uniform initialization.

```cpp
cc::util::OdbTransaction {ctx.db} ([&] {
```

Reply:
This syntax is present at 8 other locations in the codebase (authored by project **members**!) , so I assumed it is acceptable. Still think so and this level of coding style should not be target for any change request.

Reply:
Thank you for counting it. You might also want to *randomly push a change that resolves line endings* like some [other guy did to the LLVM repository recently](http://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20180730/236784.html)?

You assumed it was acceptable. Sure, I can understand it.

But now you were told it is not.

Also, project members' knowledge and code styles evolve over time, it's not like everyone is born with *x* amount of knowledge and then it stays constant. Why should **more** technical debt be added that is to be weeded out later.

Code is getting touched right now, so it should be ensured that the code that is getting touched *now* is up to some standards.
(Some standard that, may yet evolve in the future, agreeably.)

Reply:
I am not about to comply with a coding style convention you made up yourself and:
- the existing codebase does not comply with it,
- the other project members do not comply with it,
- the official coding convention of the project does not mention it.

These kind of nonsense change requests are just burning valuable research and development time for the sake of nothing.
Should be discussed on next project meeting if you disagree.

Reply:
Vexparse fixed in 4301dea623f06401b259435c3999d35a3f646dd6.